gameswap-service/src/main/java/org/gameswap/web/authentication/JwtAuthFilter.java

```diff
@@ -0,0 +1,100 @@
+package org.gameswap.web.authentication;
+
+import com.google.common.base.Optional;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jwt.JWTClaimsSet;
+
+import org.apache.commons.lang3.StringUtils;
+import org.gameswap.model.User;
+import org.gameswap.model.UserPrincipal;
+import org.gameswap.persistance.UserDAO;
+import org.gameswap.web.authentication.AuthUtils;
+import org.hibernate.CacheMode;
+import org.hibernate.FlushMode;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.context.internal.ManagedSessionContext;
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.text.ParseException;
+
+import javax.annotation.Priority;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotAuthorizedException;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.SecurityContext;
+
+@Priority(1000)
+public class JwtAuthFilter implements ContainerRequestFilter {
+    private UserDAO dao;
+    private SessionFactory sessionFactory;
+
+    public JwtAuthFilter(UserDAO dao, SessionFactory sessionFactory) {
+        this.dao = dao;
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        String authHeader = requestContext.getHeaderString(AuthUtils.AUTH_HEADER_KEY);
```

Comment:
dropwizard was doing some magic to get a hibernate session around their basic auth component. without it we can't look up the user because their is no session open.